backend/cellguide/pipeline/computational_marker_genes/computational_markers.py

```diff
@@ -317,40 +356,87 @@ def get_computational_marker_genes(self) -> dict[str, list[ComputationalMarkerGe
 
-            results = [
-                delayed(_process_cell_type__parallel)(
+            delayed_results = [
+                delayed(self._process_cell_type__parallel)(
                     i=i,
                     cell_types_o=cell_types_o,
-                    e_sum_o=e_sum_o,
-                    e_sqsum_o=e_sqsum_o,
+                    e_sum_o=e_sum_o_rollup,
+                    e_sqsum_o=e_sqsum_o_rollup,
                     n_cells_o=n_cells_o,
-                    gene_index=gene_index,
-                    combination=combination,
-                    groupby_terms=self.groupby_terms,
+                    e_sum_o_orig=e_sum_o,
+                    e_sqsum_o_orig=e_sqsum_o,
+                    n_cells_o_orig=n_cells_orig_o,
+                    filter_genes=e_nnz_o_rollup[i] < minimum_nnz,
                 )
                 for i in range(len(cell_types_o))
             ]
             logger.info(
                 f"Getting marker genes for {len(cell_types_o)} cell types in combination {combination} using {CELLGUIDE_PIPELINE_NUM_CPUS} CPUs..."
             )
-            with ProgressBar():
-                all_results.extend(compute(*results, num_workers=CELLGUIDE_PIPELINE_NUM_CPUS))
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+
+                with ProgressBar():
+                    results = compute(*delayed_results, num_workers=CELLGUIDE_PIPELINE_NUM_CPUS)
+
+                effect_sizes = []
+                for iteration in tqdm(
+                    range(len(results)),
+                    desc="Bootstrapping marker scores",
+                ):
+                    effect_sizes_chunk, col_idx = results[iteration]
+                    effects = np.full(gene_index.size, fill_value=np.nan)
+                    bootstrapped_percentiles = np.full((num_replicates, gene_index.size), fill_value=np.nan)
+
+                    if effect_sizes_chunk.shape[0] > 0:
+                        bootstrapped_percentiles[:, col_idx] = bootstrap_rows_percentiles(
+                            effect_sizes_chunk,
+                            num_replicates=num_replicates,
+                            num_samples=effect_sizes_chunk.shape[0],
+                            percentile=percentile,
+                        )
+
+                        effects[col_idx] = bootstrapped_percentiles[:, col_idx].mean(0)
+
+                    effect_sizes.append(effects)
+                effect_sizes = np.vstack(effect_sizes)
+
+                for iteration in tqdm(range(len(effect_sizes)), desc="Processing cell type marker genes"):
+                    cell_type = cell_types_o[iteration]
+
+                    effect_size = effect_sizes[iteration]
+                    not_colinear = np.array([not are_cell_types_colinear(ct, cell_type) for ct in cell_types_o])
+
+                    specificity = calculate_specificity_excluding_nans(effect_size, effect_sizes[not_colinear])
```

Comment:
exclude nans per gene and calculate the specificity of the target cell type's marker scores vs all non-colinear cell types' marker scores